### ISSUE TITLE
Idrac_bios: Unable to access the share errors

### ISSUE THREAD

**Issue by stroobl** [dell/dellemc-openmanage-ansible-modules#307]
##### Summary

I'm testing to modules to switch some new servers to uefi mode with the idrac_bios module and the only output I can get is unable to access the share. As far as I see the share is not required for many tasks and the share variables are optional.

##### Component Name
idrac_bios

##### Ansible Version
```
> ansible --version
ansible 2.10.12
  config file = <removed>/git/ansible/inventory/production/ansible.cfg
  configured module search path = ['<removed>/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/share/venv/ansible-3.4.0/lib64/python3.6/site-packages/ansible
  executable location = /usr/share/venv/ansible-3.4.0/bin/ansible
  python version = 3.6.8 (default, Mar 18 2021, 08:58:41) [GCC 8.4.1 20200928 (Red Hat 8.4.1-1)]

```

##### iDRAC or OpenManage Enterprise version
PowerEdge R740 
BIOS Version | 2.10.2
iDRAC Firmware Version | 4.40.10.00

##### Steps to Reproduce

Run a simple idrac_bios task against the drac to configure an attribute, eg:
```
  - name: Enable Uefi mode
    dellemc.openmanage.idrac_bios:
      idrac_ip: "{{ inventory_hostname }}"
      idrac_user: "{{ idrac_user }}"
      idrac_password: "{{ idrac_password }}"
      attributes:
        BootMode: Uefi
    delegate_to: localhost
```

##### Playbook used

```
- hosts: drac
  name: iDRAC playbook
  gather_facts: false
  tasks:
  - name: Enable Uefi mode
    dellemc.openmanage.idrac_bios:
      idrac_ip: "{{ inventory_hostname }}"
      idrac_user: "{{ idrac_user }}"
      idrac_password: "{{ idrac_password }}"
      attributes:
        BootMode: Uefi
    delegate_to: localhost
```

##### Expected Results
System switches from bios mode to uefi mode.

##### Actual Results
TASK [Enable Uefi mode] ***********************************************************************************
task path: <removed>/git/ansible/ansible/playbooks/drac-config.yml:54
Thursday 26 August 2021  14:56:54 +0200 (0:00:00.073)       0:00:00.073 ******* 
Using module file <removed>/.ansible/collections/ansible_collections/dellemc/openmanage/plugins/modules/idrac_bios.py
Pipelining is enabled.
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: strol
<localhost> EXEC /bin/sh -c '/usr/share/venv/ansible-3.4.0/bin/python3 && sleep 0'
The full traceback is:
  File "/tmp/ansible_dellemc.openmanage.idrac_bios_payload_8k5zva0m/ansible_dellemc.openmanage.idrac_bios_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/idrac_bios.py", line 403, in main
  File "/tmp/ansible_dellemc.openmanage.idrac_bios_payload_8k5zva0m/ansible_dellemc.openmanage.idrac_bios_payload.zip/ansible_collections/dellemc/openmanage/plugins/modules/idrac_bios.py", line 305, in run_server_bios_config
  File "/usr/share/venv/ansible-3.4.0/lib64/python3.6/site-packages/omdrivers/lifecycle/iDRAC/iDRACConfig.py", line 2675, in apply_changes
    return self._commit_scp(None, reboot=reboot)
  File "/usr/share/venv/ansible-3.4.0/lib64/python3.6/site-packages/omdrivers/lifecycle/iDRAC/iDRACConfig.py", line 2683, in _commit_scp
    content = self._sysconfig.ModifiedXML
fatal: [<removed>]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "attributes": {
                "BootMode": "Uefi"
            },
            "boot_mode": null,
            "boot_sequence": null,
            "boot_sources": null,
            "idrac_ip": "<removed>",
            "idrac_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "idrac_port": 443,
            "idrac_user": "<removed>",
            "nvme_mode": null,
            "onetime_boot_mode": null,
            "secure_boot_mode": null,
            "share_mnt": null,
            "share_name": null,
            "share_password": null,
            "share_user": null
        }
    },
    "msg": "Unable to access the share. Ensure that the share name, share mount, and share credentials provided are correct."
}

#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) 
  to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, 
  they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by anupamaloke** [comment]
@stroobl,  I am guessing this could be due to the user permission issue. By default, the module tries to the use the ```tempfile.gettempdir()``` for storing the temporary SCP file before importing it to the iDRAC.   It seems that the linux username you are using does not have permissions to access the ```/tmp``` directory.

```
>>> import tempfile
>>> print(tempfile.gettempdir())
/tmp
```

Could you please try with the following small modification to your existing playbook? With this modification, I am explicitly telling the module to use the current playbook directory to store the temporary SCP files instead of using the default ```/tmp```.
```
- hosts: drac
  name: iDRAC playbook
  gather_facts: false
  tasks:
  - name: Enable Uefi mode
    dellemc.openmanage.idrac_bios:
      idrac_ip: "{{ inventory_hostname }}"
      idrac_user: "{{ idrac_user }}"
      idrac_password: "{{ idrac_password }}"
      share_name: "{{ playbook_dir }}"              <======= Add this
      attributes:
        BootMode: Uefi
    delegate_to: localhost
 ```

**Comment by stroobl** [comment]
The problem stays the same when I add the share_name (and I can write in /tmp):
```
> touch /tmp/brol
 > echo $?
0
```

**Comment by anupamaloke** [comment]
@stroobl, I think I might have found as to why it is not working. Could you please open the file ```/.ansible/collections/ansible_collections/dellemc/openmanage/plugins/modules/idrac_bios.py``` in a editor and comment out the following lines 263-265:

```
    if not upd_share.IsValid:
        module.fail_json(msg="Unable to access the share. Ensure that the share name, "
                             "share mount, and share credentials provided are correct.")
```

Once you do that, save the file and re-run the playbook. Let me know if you still see the issue.

**Comment by stroobl** [comment]
@anupamaloke Still exactly the same error. Maybe I end up on the same message at line 417?

**Comment by anupamaloke** [comment]
@stroobl, may be you are right. A last throw of dice from my end. Could you please may be try with a new python virtual environment and see if this error this persists? If yes, then it would be good to open a SR with us. 

Also, what is the version of the ```omsdk``` python library that is installed in your environment?
```
$ pip freeze | grep omsdk
omsdk==1.2.478
```
I just tried in my environment by configuring a BIOS attribute and it worked

**Comment by stroobl** [comment]
I'm on that omsdk version: omsdk==1.2.478. Only started testing recently with these modules. I'll try a new virtualenv, but I don't expect it to help...

Could it be related to the fact that we don't have full OME access? Our user is somewhat limited since we only manage a couple of servers from this larger OME setup.  I am able to switch from BIOS to UEFI and set PXE configuration in the web interface, but I noticed that I don't have access to export and re-import server configuration profiles.

**Comment by anupamaloke** [comment]
> Could it be related to the fact that we don't have full OME access? Our user is somewhat limited since we only manage a couple of servers from this larger OME setup. I am able to switch from BIOS to UEFI and set PXE configuration in the web interface, but I noticed that I don't have access to export and re-import server configuration profiles.

@stroobl, oh yes. This completely escaped me when I responded last. You must have permissions to export and import SCP. Alternatively, you can use the redfish modules from the ```community.general``` collection:

```
  vars:
    ansible_python_interpreter: "/usr/bin/env python"
    bios_attributes:
      BootMode: "Uefi"

  tasks:
  - name: Update bios attributes
    community.general.redfish_config:
      baseuri: "{{ inventory_hostname }}"
      username: "{{ user }}"
      password: "{{ password }}"
      category: "Systems"
      resource_id: "System.Embedded.1"
      command: "SetBiosAttributes"
      bios_attributes:
        BootMode: "Uefi"
    register: update_bios_attr_resp

  - name: Create bios configuration job
    community.general.idrac_redfish_command:
      baseuri: "{{ inventory_hostname }}"
      username: "{{ user }}"
      password: "{{ password }}"
      category: "Systems"
      command: "CreateBiosConfigJob"
    register: bios_config_job_resp
    when:
      - update_bios_attr_resp.changed

  - name: Reboot server to start BIOS configuration job
    community.general.redfish_command:
      baseuri: "{{ inventory_hostname }}"
      username: "{{ user }}"
      password: "{{ password }}"
      category: "Systems"
      resource_id: "System.Embedded.1"
      command: "PowerGracefulRestart"
    when:
      - bios_config_job_resp.changed
```

**Comment by stroobl** [comment]
Thanks for the info, I'll give that a try. I was not aware that I could use that redfish module for bios configuration (I already found out about the redfish module for the storage config included in the collection). I think it would be good to add some pointers on which modules to use for which use case in the documentation.

**Comment by vzovko** [comment]
I'm have the same issue as @stroobl but the workaround mentioned by @anupamaloke in https://github.com/dell/dellemc-openmanage-ansible-modules/issues/307#issuecomment-914071063 is working fine. In my case I'm disabling the internal SD card.

**Comment by scottgiff** [comment]
I ran across this issue as well. I found in my case there was an orphaned job in the idrac jobqueue that was blocking the BIOS attribute update.  I had to do a racadm jobqueue -i JID_CLEARALL_FORCE and then the error was gone.

**Comment by anupamaloke** [comment]
Closed by #445
